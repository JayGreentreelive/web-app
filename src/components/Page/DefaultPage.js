import React from 'react'
import PropTypes from 'prop-types'
import {
  useQuery
} from 'react-apollo'
import {
  mapEdgesToNodes,
} from '../../utils'
import getWebPageBlockItems from '../../queries/getWebPageBlockItems'

import SEO from '../../seo'
import PixelManager from '../PixelManager'

import { Loader } from '@christfellowshipchurch/web-ui-kit'
import ContentBlock from '../ui/ContentBlock'
import GroupBlock from '../ui/GroupBlock'


const DefaultPage = ({ title, match: { params: { page } } }) => {
  PixelManager.initWithPageView(`/${page || ''}`)

  const website = process.env.REACT_APP_WEBSITE_KEY
  const { loading, error, data } = useQuery(getWebPageBlockItems, { variables: { website, title: page || title } })

  if (loading) return (
    <div className="vh-100 vw-100 d-flex justify-content-center align-items-center bg-light">
      <Loader />
    </div>
  )

  if (error) {
    console.error("ERROR: ", error)
    return <h1 className="text-center">There was an error loading the page. Please try again.</h1>
  }

  const blockItems = mapEdgesToNodes(data.getWebsitePageContentByTitle.childContentItemsConnection)
  const {
    metaDescription,
    metaKeywords,
    openGraphProtocols,
    twitterProtocols,
  } = blockItems

  return (
    <>
      <SEO
        title={`${page || title} - Christ Fellowship Church`}
        description={metaDescription}
        keywords={metaKeywords}
        openGraphProtocols={openGraphProtocols}
        twitterProtocols={twitterProtocols}
      />

      {blockItems.map((item, i) => {
        const classenames = i % 2 === 0 ? 'bg-white' : 'bg-light'
        switch (item.__typename) {
          case 'WebsiteBlockItem':
            return (
              <div className={classenames} key={i}>
                <ContentBlock {...item} />
              </div>
            )
          case 'WebsiteGroupItem':
            return (
              <div className={classenames} key={i}>
                <GroupBlock {...item} />
              </div>
            )
          default:
            return <h1 className="text-center" key={i}>{item.title}</h1>
        }
      })}
    </>
  )
}

DefaultPage.defaultProps = {
  title: 'home',
}

DefaultPage.propTypes = {
  title: PropTypes.string,
  match: PropTypes.shape({
    params: PropTypes.shape({
      page: PropTypes.string
    })
  })
}

export default DefaultPage
