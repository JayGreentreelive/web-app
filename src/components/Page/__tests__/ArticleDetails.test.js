import React from "react"
import {
  Accordion,
  Carousel,
  Row,
  Loader,
  Block,
  Media,
  Button
} from "@christfellowshipchurch/web-ui-kit"
import { ContentBlock, GroupBlock } from '../DefaultPage'
import renderer from "react-test-renderer"

describe("ArticleDetails", () => {

  it("renders without crashing", () => {
    const tree = renderer.create(
     <Block>
       <div>
        <Block.Title></Block.Title>
        <Block.Subtitle></Block.Subtitle>
        <Media
          imageUrl="url"
          imageAlt="alt"
        />
          <div>
            <Media
              imageUrl="url"
              imageAlt="alt"
            />
              <div>
                <p></p>
                <p></p>
              </div>
          </div>
        <Block.Body></Block.Body>
       </div>    
     </Block>
    )
    expect(tree).toMatchSnapshot()
  })
});
