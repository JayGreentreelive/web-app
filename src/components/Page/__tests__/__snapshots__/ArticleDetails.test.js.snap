// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetails renders without crashing 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    />
    <div
      className="col-12 col-md text-center px-3"
    >
      <div>
        <h2 />
        <h6 />
        <div
          className="embed-responsive embed-responsive-1by1  "
        />
        <div>
          <div
            className="embed-responsive embed-responsive-1by1  "
          />
          <div>
            <p />
            <p />
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`DefaultPage renders without crashing 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    />
    <div
      className="col-12 col-md text-center px-3"
    >
      <div>
        <h2 />
        <h6 />
        <div
          className="embed-responsive embed-responsive-1by1  "
        />
        <div>
          <div
            className="embed-responsive embed-responsive-1by1  "
          />
          <div>
            <p />
            <p />
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
